assets/app/view/game/auto.rb

```diff
@@ -336,6 +336,67 @@ def render_share_pass(settings)
         children
       end
 
+      def enable_independent_mines(form)
+        settings = params(form)
+
+        skip_track = settings['skip_track']
+        skip_buy = settings['skip_buy']
+        skip_close = settings['skip_close']
+        indefinite = settings['indefinite']
+
+        process_action(
+          Engine::Action::ProgramIndependentMines.new(
+            sender,
+            skip_track: skip_track,
+            skip_buy: skip_buy,
+            skip_close: skip_close,
+            indefinite: indefinite
+          )
+        )
+      end
+
+      def render_checkbox(label, id, form, checked)
+        render_input(
+          label,
+          id: id,
+          type: 'checkbox',
+          inputs: form,
+          attrs: {
+            checked: checked,
+          }
+        )
+      end
+
+      def render_independent_mines(settings)
+        form = {}
+        text = 'Auto Independent Mines'
+        text += ' (Enabled)' if settings
+        children = [h(:h3, text)]
+        children << h(:p,
+                      'Automatically skip independent mine actions.'\
+                      ' This will deactivate itself in the next SR, unless set to indefinite.'\
+                      ' It will also deactivate itself when a mine has negative income.')
+
+        skip_track = settings ? settings&.skip_track : true
```

Comment:
you could remove all of these and just do 

render_checkbox(... settings&.skip_track : true)